force-app/main/default/classes/RecordTypeService.cls

```diff
@@ -86,6 +86,30 @@ public virtual with sharing class RecordTypeService {
         return activeRecordTypeModelsById;
     }
 
+    /***********************************************************************************************************************
+     * @description Filter through a map of valid Record Type models to retrieve a list of Ids corresponding to values in a specified list.
+     * @param  idsToFindList A list of Ids corresponding to Record Type models
+     * @param  validRecordTypeModelsById A map of valid Record Type models by Id
+     * @return A List<Id> containing Ids that correspond to valid Record Type models and match a value in the specified list of Ids to find.
+     ***********************************************************************************************************************/
+    public virtual List<Id> filterListForValidIds(
+        List<Id> idsToFindList,
+        Map<Id, RecordTypeModel> validRecordTypeModelsById
+    ) {
+        List<Id> validIdsList = new List<Id>();
+
+        for (Id idToFind : idsToFindList) {
+            if (validRecordTypeModelsById.containsKey(idToFind) == false) {
```

Comment:
Considering this implementation, we could use `getRecordTypeModelFromMap` for consistent and reusable logic. @xinnanlisf, how do you feel about making this method lighter weight and relying on a `List<String>` and `Set<Id>` vs using a `List<String>` and `Map<Id,RecordType>` with `getRecordTypeModelFromMap`?

Reply:
I prefer the lighter weight version, it seems more straightforward to me

Reply:
Make it so!